### ISSUE TITLE
Enhancement: [restrict-plus-operands] no error for concat with empty string (typecast)

### ISSUE THREAD

**Issue by kkmuffme** [typescript-eslint/typescript-eslint#7035]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/restrict-plus-operands/

### Description

Concating a number with an empty string is a commonly used way to change a number to a string.

This rule (maybe optionally via a config option) shouldn't report that.

### Fail

```typescript
no change
```

### Pass

```typescript
const pass = 15 + '';
```

**Comment by bradzacher** [comment]
I'm not sure I understand why this case should be ignored - why?

There are many other options to do this without a `+`, for example:
- `(15).toString()`
- `15..toString()`
- `15..toFixed(0)`
- `String(15)`
- `` `${15}` ``

My response to #7034 applies here as well - there are a lot of cases where implicitly coercing a number to a string is not the correct output, and instead a different construct (like `.toFixed`) should be used - so why should this be special cased?
